test/spitfire_test.exs

```
@@ -534,6 +534,24 @@ defmodule SpitfireTest do
       end
     end
 
+    test "unfinished keyword list" do
+      code = ~S'''
+      defmodule MyModule do
+        IO.inspect(
+          :stderr,
+          label: "label",
+          (__cursor__())
+        )
+      end
+      '''
```

this is an interesting result from the builtin parser

```
iex(5)> Code.string_to_quoted(~S'''
...(5)>       defmodule MyModule do
...(5)>         IO.inspect(
...(5)>           :stderr,
...(5)>           label: "label",
...(5)>           (__cursor__())
...(5)>         )
...(5)>       end
...(5)>       ''')
{:ok,
 {:defmodule, [line: 1],
  [
    {:__aliases__, [line: 1], [:MyModule]},
    [
      do: {{:., [line: 2], [{:__aliases__, [line: 2], [:IO]}, :inspect]},
       [line: 2], [:stderr, [{:label, "label"}, {:__cursor__, [line: 5], []}]]}
    ]
  ]}}
iex(6)> Code.string_to_quoted(~S'''
...(6)> defmodule MyModule do
...(6)>   IO.inspect(
...(6)>     :stderr,
...(6)>     label: "label",
...(6)>     :foo
...(6)>   )
...(6)> end
...(6)> '''
...(6)> )
{:error,
 {[line: 4, column: 19],
  "unexpected expression after keyword list. Keyword lists must always come as the last argument. Therefore, this is not allowed:\n\n    function_call(1, some: :option, 2)\n\nInstead, wrap the keyword in brackets:\n\n    function_call(1, [some: :option], 2)\n\nSyntax error after: ",
  "','"}}
iex(7)>
```

---

Either the builtin parser supports a macro call returning tuple AST node or this is an error

---

Another possibility it is more forgiving for nodes with `__cursor__()` special form

---

@mhanberg We have two options here.
1. Have a special handling for `__cursor__()` nodes in keyword lists
2. Make the parser error tolerant and either silently drop invalid list entries or produce invalid AST